File: Data Structures & Algorithms/climbing-stairs/submission-1.java
class Solution {
    public int climbStairs(int n) {
        if(n == 1){
            return 1;
        }else if (n==2){
            return 2;
        }
        int[] T = new int[n];

        T[0] = 1;
        T[1] = 2;
        for (int i = 2; i< n; i++){
            T[i] = T[i-1] + T[i-2];
        }
        return T[n-1];
    }
}

File: Data Structures & Algorithms/last-stone-weight/submission-1.py
class Solution:
    def lastStoneWeight(self, stones: List[int]) -> int:
        '''
        input : stones array

        for each step : select two heavy stone
            if x == y, remove both
            else smaller weight distroied , larger one weight updated to y-x
        repeat this process till only one stone remains

        '''

        heapq.heapify_max(stones)
        while(len(stones)>1):
            x = heapq.heappop_max(stones)
            y = heapq.heappop_max(stones)
            if(x==y):
                continue
            else:
                z = abs(x-y)
                heapq.heappush_max(stones,z)
            print(stones)
        if len(stones) == 0:
            return 0
        return stones[0]